### PR TITLE
DRM-1  - create components for Home Page layout (Header, Description, Grid). Create HomePage component. 

### PR DESCRIPTION
**Pull request by OlgaMilczek**
Changes: 
- App.txs - Add StylesProvider injectFirst to fix problem with overwriting styled-components by Material UI. 
- theme.ts - Add breakpoints for , primary color for text, default Font-family. Change font-size for h1 (default 6em). 
Adds: 
- MainContainer.tsx ( Main responsive container with background img, common to all pages)
- Header.tsx (Home page header wit main title)
- Description.tsx
- GridContainer.tsx - container for specialization cards. 
- HomePage.tsx - HomePage main component. (to fill by rest of the content).

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-    <ThemeProvider theme={theme}>
-      {/* Replace with actual content */}
-      <DemoContainer />
-    </ThemeProvider>
-  </MuiThemeProvider>
+  <StylesProvider injectFirst>
+    <MuiThemeProvider theme={theme}>
+      <ThemeProvider theme={theme}>
+        {/* Replace with actual content */}
+        <DemoContainer />
```
Comment (klaraborowska) [inline src/App.tsx]: You can replace `DemoContainer` with your `HomePageLayout` :)

File: src/layout/components/Description.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import Typography from '@material-ui/core/Typography';
+import styled from 'styled-components';
+
+const DescriptionContainer = styled('div')`
+  text-align: center;
+  margin: auto;
+  width: 80%;
+  margin: auto;
```
Comment (klaraborowska) [inline src/layout/components/Description.tsx]: This style is duplicated (2 lines above).

File: src/layout/components/Description.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        margin-bottom: .8em;
+      }
+  }
+`;
+
+export const Description: React.FC<{}> = () => (
+  <DescriptionContainer>
+    <Typography color="textPrimary">So you want to work in IT? Or you just curious about opportunities in the TECH world?</Typography>
+    <Typography color="textPrimary">You&apos;re in the right place!</Typography>
+    <Typography color="textPrimary">Explore the options below and decide which path works best for you.</Typography>
```
Comment (klaraborowska) [inline src/layout/components/Description.tsx]: You could add this style for `color` to body, as we want to have the same primary text color for whole app. This style is inherited by all children, so you could remove it from here than.

File: src/layout/components/GridContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+/* Grid container for the main page */
+export const GridContainer = styled('div')`
+    max-width: 960px;
+    width: 80%;
+    display: grid;
+    margin: auto; 
+    margin-top: 2em;
+    grid-template-columns: repeat(auto-fit, minmax(min(350px, 100%), 1fr));
+    grid-column-gap: 2em;
+    grid-row-gap: 2em;
```
Comment (klaraborowska) [inline src/layout/components/GridContainer.tsx]: There is a style `grid-gap` which includes both `grid-column-gap` and `grid-row-gap`.

File: src/layout/components/MainContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+
+/* Main container with background img for all pages */
+export const MainContainer = styled('div')`
+    background-image: url('./img/background-img.png');
+    background-size: cover; 
+    background-position: center top;
+    background-repeat: no-repeat;
```
Comment (klaraborowska) [inline src/layout/components/MainContainer.tsx]: Do you know that there is a shortcut for background styles? Of course your styles are just fine, but if you want you can replace them with `background: url('./img/background-img.png') center top / cover no-repeat;`. The only thing to remember is to separate `background-size` and `background-position` with a slash.

Comment (OlgaMilczek) [reply src/layout/components/MainContainer.tsx]: I know :) It is more personal preference, this way is more readable for me (probably because of lack of experience). But maybe I should used to sortcut.

File: src/layout/components/GridContainer.tsx (hunk 4 above)
Comment (OlgaMilczek) [reply src/layout/components/GridContainer.tsx]: Done :)

File: src/layout/components/Description.tsx (hunk 3 above)
Comment (OlgaMilczek) [reply src/layout/components/Description.tsx]: I should add color to index.css or there is some way to do it with a theme? Because I am looking for some solution, but I can't find it.

Comment (OlgaMilczek) [reply src/layout/components/Description.tsx]: I find something like this:

```
const theme = createMuiTheme({
  typography: {
    allVariants: {
      color:  '#222222'
    },
  },
});
```

Is this a right solution?

File: src/App.tsx (hunk 1 above)
Comment (OlgaMilczek) [reply src/App.tsx]: Done!

File: src/layout/components/Description.tsx (hunk 2 above)
Comment (OlgaMilczek) [reply src/layout/components/Description.tsx]: Done!

File: src/layout/components/Description.tsx (hunk 3 above)
Comment (klaraborowska) [reply src/layout/components/Description.tsx]: Looks good, if it's working than I think it's fine :) Just check, if it's also working for other elements than material typography, for example `<div>` or `<p>`.  If it's not working  than I would just add this style to `index.css` to `body`.

Comment (OlgaMilczek) [reply src/layout/components/Description.tsx]: I check and it actually doesn't work for plain jsx elements, so I add color and font-family to index.css to body and that's work just fine.

**Review by klaraborowska — APPROVED** [review]

**Review by joannajerz — APPROVED** [review]

File: src/home-page/components/GridContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+/* Grid container for the main page */
+export const GridContainer = styled('div')`
+    max-width: 960px;
+    width: 80%;
+    display: grid;
+    margin: auto; 
+    margin-top: 2em;
+    grid-template-columns: repeat(auto-fit, minmax(min(350px, 100%), 1fr));
+    grid-gap: 2em;
+    justify-items: center;
```
Comment (vilanelle) [inline src/home-page/components/GridContainer.tsx:12]: Maybe `justify-items: stretch;` would be a better fit for our needs? :) I don't have much experience with `display: grid;` so feel free to ignore this comment if I'm missing something obvious here :D

File: src/home-page/components/GridContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import styled from 'styled-components';
```
Comment (vilanelle) [inline src/home-page/components/GridContainer.tsx:1]: Just a suggestion or food for thought: Right now we only use this nice grid container for `HomePage`, but it could be easily reused in other places of the app if needed. What do you think about moving it to a shared folder? But leaving it here for now (as long as we don't reuse it elsewhere) is perfectly fine, too :)

File: src/index.css [hunk 8]
```diff
@@ -1,10 +1,9 @@
 body {
```
Comment (vilanelle) [inline src/index.css:1]: We could move these global styles to styled components: https://styled-components.com/docs/api#createglobalstyle
That way, we'll be able to use our theme easily for global styles. Of course, this is out of scope of this PR - I added a separate task in Jira: https://dareit.atlassian.net/browse/DRM-35

File: src/App.tsx [hunk 9]
```diff
@@ -1,14 +1,17 @@
...
-  <MuiThemeProvider theme={theme}>
-    <ThemeProvider theme={theme}>
-      {/* Replace with actual content */}
-      <DemoContainer />
-    </ThemeProvider>
-  </MuiThemeProvider>
+  <StylesProvider injectFirst>
+    <MuiThemeProvider theme={theme}>
+      <ThemeProvider theme={theme}>
+        {/* Replace with actual content */}
```
Comment (vilanelle) [inline src/App.tsx]: I guess we can remove this comment now :)

**Review by vilanelle — APPROVED** [review]

File: src/index.css (hunk 8 above)
Comment (OlgaMilczek) [reply src/index.css:1]: Ok. I can do it. :)

File: src/App.tsx (hunk 9 above)
Comment (OlgaMilczek) [reply src/App.tsx]: Ok.

File: src/home-page/components/GridContainer.tsx (hunk 6 above)
Comment (OlgaMilczek) [reply src/home-page/components/GridContainer.tsx:12]: Hmm... I used center because when it is too narrow for two elements it is still too wide for one - in my opinion. But i can add center to media queries for tablets, and for desktop and phone it can be stretch or we can change it when we will see how it works with actual card inside.
